### PR TITLE
CI: Test on Python 3.7 on Travis and AppVeyor

### PR DESCRIPTION
**Pull request by effigies**
Hopefully will reproduce #650.

### CODE REVIEW

**Comment by effigies** [comment]
There doesn't currently seem to be any way to test on Python 3.7 on Travis. The suggestions from travis-ci/travix-ci#9815 fail.

**Comment by matthew-brett** [comment]
Maybe install pyenv ?  I think that has Python 3.7 available.

**Comment by effigies** [comment]
It looks like pyenv is pre-installed, but the docs aren't clear on how to use that to install and use a specific version. Can do some digging, or is this something you're familiar enough with to do quickly>

**Comment by grlee77** [comment]
You could also try specifying python as `3.7-dev`. That uses a beta version of 3.7, which may be good enough until 3.7 is available on Travis? That is what we have for now in PyWavelets/pywt#390.

**Comment by effigies** [comment]
@grlee77 Looks like that was enough to reproduce the issue. Thanks.

**Comment by effigies** [comment]
Okay, I've had a look at this, and I don't understand warning registries enough to be sure I'd be resolving it correctly. In 3.7, this code:

https://github.com/nipy/nibabel/blob/bb8c6fa79e5ac8430d25b66a70bd1175ed67d673/nibabel/tests/test_testing.py#L86-L88

Produces `my_mod.__warningregistry__ == {'version': 14}`. So the test is kind of correctly failing, in that our expectations are violated. But I don't know how to respond.

**Comment by effigies** [comment]
The `warnings.simplefilter('ignore')` lines were preventing warnings from being added to registries. It appears that this is a change in 3.7, where the registry is not filled in these cases, though whether that's considered a fixed bug or changed behavior is unclear to me. In any event, it doesn't seem relevant to our tests, so I removed it.

**Comment by effigies** [comment]
@matthew-brett Care to review?

**Comment by effigies** [comment]
When this goes in, I think it would be good to cut a bug-fix release.

**Comment by effigies** [comment]
If there's no objection, I'll merge Monday. It would be good to get 3.7 testing for all new PRs.

File: .travis.yml [hunk 1]
```diff
@@ -22,11 +21,15 @@ env:
         - EXTRA_PIP_FLAGS="--find-links=$EXTRA_WHEELS"
         - PRE_PIP_FLAGS="--pre $EXTRA_PIP_FLAGS --find-links $PRE_WHEELS"
 python:
-    - 3.4
+    - 2.7
```
Comment (matthew-brett) [inline .travis.yml]: Why the extra 2.7 test?

Comment (effigies) [reply .travis.yml]: I think it looked like we were missing the `INSTALL_TYPE="setup"` from the `INSTALL_TYPE` block. But I guess all of them are hitting that, so I'll drop it.

**Comment by matthew-brett** [comment]
Great - thanks - in it goes.

**Pull request merged**